app/javascript/packages/document-capture/context/hijack-skip-nav.tsx

```diff
@@ -0,0 +1,22 @@
+/**
+ * On pages that use the # value for navigation, we need to intercept
+ * and alter the functionality of the 'Skip to main content' link.
+ * This function prevents the link from appending the hash to the url,
+ * and instead it focuses on the main-content section of the page.
+ * When the React FSM navigation is changed so that we no longer use the # value,
+ * we should remove this code.
+ * @param - null
+ *
+ * @return - null
+ */
+
+export default function hijackSkipNav(): void {
+  const skipNavLink = document.querySelector<HTMLFormElement>('.usa-skipnav');
+  const mainContent = document.querySelector<HTMLFormElement>('#main-content');
+
+  skipNavLink?.addEventListener('click', (e) => {
+    e.preventDefault();
+    mainContent?.focus();
+    mainContent?.scrollIntoView();
+  });
+}
```

Comment:
Here's an example of the kind of logic I mentioned.

```suggestion
// Function has same identity between calls, so can't be duplicated
function hijackListener(e: Event) {
  e.preventDefault();
  
  // Get the main content on click so we don't need to worry about it being removed;
  // this is an infrequent operation, so a repeated DOM search is acceptable
  const mainContent = document.querySelector<HTMLFormElement>('#main-content');
  mainContent?.focus();
  mainContent?.scrollIntoView();
}

// Keep track of the listener calls so we don't cleanup the listener until it's appropriate
// Listeners are no longer relevant after there's no reference to the element, so WeakMap is appropriate
const listenerCounts = new WeakMap<HTMLFormElement,number>();

export default function hijackSkipNav(): () => void {
  const skipNavLink = document.querySelector<HTMLFormElement>('.usa-skipnav');

  skipNavLink?.addEventListener('click', hijackListener);
  
  const prevCnt = listenerCounts.get(skipNavLink) || 0;
  listenerCounts.set(skipNavLink, prevCnt + 1);
  
  // Return a function that can be passed as the useEffect return value for listener cleanup
  return () => {
    // Only cleanup if this is the last usage; otherwise decrement the listener counter
    const cnt = listenerCounts.get(skipNavLink);
    if (cnt === 1) {
      listenerCounts.delete(skipNavLink);
      skipNavLink?.removeEventListener('click', hijackListener);
    } else if (cnt > 1) {
      listenerCounts.set(skipNavLink, cnt - 1);
    }
  };
}
```

Reply:
Could be overkill since this should only be happening once per page.

Reply:
Ok, this helps clarify some things! I think first things first we should decide if we're following @aduth's advice of implementing this as the default behavior, if so then no need to wrangle with the `useEffect` block. @NavaTim do you have an opinion about making this the default behavior?

Reply:
If @aduth said to make this the default behavior, then I'm good with that.